test/explain/algoritms/test_gnn_explainer.py

```diff
@@ -0,0 +1,383 @@
+        self.conv2 = GCNConv(16, 16)
+        self.lin = Linear(16, 7)
+
+    def forward(self, x, edge_index, edge_attr=None, batch=None):
+        # edge_attr is for checking if explain can
+        # pass arguments in the right oder
+        x = self.conv1(x, edge_index)
+        x = F.relu(x)
+        x = self.conv2(x, edge_index)
+        x = global_add_pool(x, batch)
+        return self.lin(x)
+
+
+class GNN_regression_singleoutput(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = GCNConv(3, 16)
+        self.conv2 = GCNConv(16, 16)
+        self.lin = Linear(16, 1)
+
+    def forward(self, x, edge_index, edge_attr=None, batch=None):
+        # edge_attr is for checking if explain can
+        # pass arguments in the right oder
+        x = self.conv1(x, edge_index)
+        x = F.relu(x)
+        x = self.conv2(x, edge_index)
+        x = global_add_pool(x, batch)
+        return self.lin(x)
+
+
+class GNN_classification_singleoutput(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = GCNConv(3, 16)
+        self.conv2 = GCNConv(16, 16)
+        self.lin = Linear(16, 7)
+
+    def forward(self, x, edge_index, edge_attr=None, batch=None):
+        # edge_attr is for checking if explain can
+        # pass arguments in the right oder
+        x = self.conv1(x, edge_index)
+        x = F.relu(x)
+        x = self.conv2(x, edge_index)
+        x = global_add_pool(x, batch)
+        x = self.lin(x)
+        return x.log_softmax(dim=1)
+
+
+class GNN_classification_multioutput(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = GCNConv(3, 16)
+        self.conv2 = GCNConv(16, 16)
+        self.lin = Linear(16, 7)
+        self.lin2 = Linear(16, 7)
+
+    def forward(self, x, edge_index, edge_attr=None, batch=None):
+        # edge_attr is for checking if explain can
+        # pass arguments in the right oder
+        x = self.conv1(x, edge_index)
+        x = F.relu(x)
+        x = self.conv2(x, edge_index)
+        x = global_add_pool(x, batch)
+        out_1 = self.lin(x).log_softmax(dim=1)
+        out_2 = self.lin2(x).log_softmax(dim=1)
+        return torch.stack([out_1, out_2], dim=0)
+
+
+# --------------------------------------------------------------------
+# check explanations
+# --------------------------------------------------------------------
+def check_explanation(
+    edge_mask_type,
+    node_mask_type,
+    x,
+    edge_index,
+    explanation,
+):
+    if node_mask_type == MaskType.attributes:
+        assert explanation.node_features_mask.shape == x.shape
+        assert explanation.node_features_mask.min() >= 0
+        assert explanation.node_features_mask.max() <= 1
+    elif node_mask_type == MaskType.object:
+        assert explanation.node_mask.shape == x.shape[0]
+        assert explanation.node_mask.min() >= 0
+        assert explanation.node_mask.max() <= 1
```

Comment:
Maybe also check expected behavior when the mask is None, in addition to `common_attributes` and `both`.

Reply:
The initial  method does not support the `None` attributes for the node mask, I'll update it in the docs.

https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/models/gnn_explainer.py#L49-L53